crates/uniffi-bindgen-react-native/src/bindings/react_native/gen_typescript/templates/ObjectTemplate.ts

```diff
@@ -84,11 +83,35 @@ export class {{ impl_class_name }} implements {{ protocol_name }}, UniffiObjectI
 
 }
 
-function create{{ impl_class_name }}(pointer: UnsafeMutableRawPointer): {{ impl_class_name }} {
-    const instance = Object.create({{ impl_class_name }}.prototype);
-    instance.pointer = pointer;
-    return instance;
-}
+const {{ obj_factory }}: UniffiObjectFactory<{{type_name}}> = {
+    create(pointer: UnsafeMutableRawPointer): {{ type_name }} {
+        const instance = Object.create({{ type_name }}.prototype);
+        instance._rustArcPtr = this.bless(pointer);
+        return instance;
+    },
+
+    bless(pointer: UnsafeMutableRawPointer): UniffiRustArcPtr {
+        const destructor = this.freePointer;
```

Comment:
Not totally sure, but this looks like it might need to be 'bound' to the this pointer to function correct, if its going to be called as a standalone callback function:

`bind(this, this.freePointer)`

Unless the `this` is going to be injected as the first argument when it is called by `uniffiBlessPointer`, otherwise the this object will be missing.

Reply:
I might be misunderstanding, it just has the smell of something that needs a `bind`